core/src/main/java/bisq/core/trade/asset/xmr/XmrProofInfo.java

```diff
@@ -0,0 +1,152 @@
+                    log.warn("txHash {}, expected: {}", jsonTxHash.getAsString(), txHash);
+                    return new XmrProofResult(XmrProofResult.State.TX_HASH_INVALID, null);
+                }
+            }
+
+            // validate that the txkey matches
+            final JsonElement jsonViewkey = jsonData.get("viewkey");
+            if (jsonViewkey == null) {
+                return new XmrProofResult(XmrProofResult.State.API_INVALID, "missing viewkey field");
+            } else {
+                if (!jsonViewkey.getAsString().equalsIgnoreCase(this.txKey)) {
+                    log.warn("viewkey {}, expected: {}", jsonViewkey.getAsString(), txKey);
+                    return new XmrProofResult(XmrProofResult.State.TX_KEY_INVALID, null);
+                }
+            }
+
+            // calculate how many confirms are still needed
+            int confirmations = 0;
+            final JsonElement jsonConfs = jsonData.get("tx_confirmations");
+            if (jsonConfs == null) {
+                return new XmrProofResult(XmrProofResult.State.API_INVALID, "missing tx_confirmations field");
+            } else {
+                confirmations = jsonConfs.getAsInt();
+                log.info("confirmations: {}, xmr txid: {}", confirmations, txHash);
+            }
+
+            // iterate through the list of outputs, one of them has to match the amount we are trying to verify.
+            // check that the "match" field is true as well as validating the amount value
+            final JsonArray jsonOutputs = jsonData.get("outputs").getAsJsonArray();
+            for (int i = 0; i < jsonOutputs.size(); i++) {
+                JsonObject out = jsonOutputs.get(i).getAsJsonObject();
+                if (out.get("match").getAsBoolean()) {
+                    long jsonAmount = out.get("amount").getAsLong();
+                    if (jsonAmount == amount) {
+                        if (confirmations < confirmsRequired)
+                            // we return TX_NOT_CONFIRMED which will cause a retry later
+                            return new XmrProofResult(confirmations, confirmsRequired, XmrProofResult.State.TX_NOT_CONFIRMED);
+                        else
+                            return new XmrProofResult(confirmations, confirmsRequired, XmrProofResult.State.PROOF_OK);
+                    }
+                }
+            }
+
+            // reaching this point means there was no matching amount
+            return new XmrProofResult(XmrProofResult.State.AMOUNT_NOT_MATCHING, null);
+
+            // TODO check if date of tx is after tradeDate (allow some tolerance to avoid clock sync issues).
+            //  Otherwise a scammer could send an old txKey from a prev trade with same amount before seller has
+            //  updated to new feature, thus the check for duplication would not detect the scam.
+            //  >>>>> the API does not provide date of tx <<<<<<
```

Comment:
@chimp1984 how important is it to get tradeDate from the monero API, as it is not currently there?

Reply:
I think its just some edge case at update where this could be a risk. Does the API not provide the date? If not would you mind to ask the Monero dev if he can add it? I guess it would be a small change and would remove the described risk.